sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/AnsiTypeCoercion.scala

```diff
@@ -182,16 +184,19 @@ object AnsiTypeCoercion extends TypeCoercionBase {
 
       // If a function expects a StringType, no StringType instance should be implicitly cast to
       // StringType with a collation that's not accepted (aka. lockdown unsupported collations).
-      case (_: StringType, StringType) => None
+      case (_: StringType, _: StringType) => None
       case (_: StringType, _: StringTypeCollated) => None
 
+      case (DateType, AnyTimestampType) =>
```

Comment:
why do we need this change? It's not even related to string type.

Reply:
Let me check, this is a problem I needed to solve as a conflict with another PR, which simplified this function